backend/router.ts

```diff
@@ -0,0 +1,82 @@
+import { Router, Request, Response} from "express";
+import {DatabaseFunctions} from "./repository";
+import {createHmac} from "crypto";
+
+export class MainRouter {
+    public readonly router: Router = Router()
+
+    constructor() {
+        this.router = Router()
+        this.createRoutes()
+    }
+
+    private createRoutes() {
+        this.router
+            .get('/orders', this.getOrders)
+            .post('/search', this.searchItem)

```

Comment:
Someone might say that you should use GET here instead of POST. In my opinion if search end-point doesn't take a lot of arguments and wouldn't be too long (in this case it takes only query text, minPrice and minPrice, so probably wouldn't) then we should stick to GET with query params.